src/app/subnets/views/VLANDetails/VLANDetails.tsx

```diff
@@ -34,7 +34,7 @@ const VLANDetails = (): JSX.Element => {
   );
   const vlansLoading = useSelector(vlanSelectors.loading);
   const subnets = useSelector((state: RootState) =>
-    subnetSelectors.getByVLAN(state, id)
+    subnetSelectors.getByIds(state, vlan ? vlan.subnet_ids : [])
```

Comment:
Using an optional chaining operator would be both shorter and safer (as it would result in a `[]` fallback even if `subnet_ids` have a falsy value).

```suggestion
    subnetSelectors.getByIds(state, vlan?.subnet_ids || [])
```

Reply:
nice spot!